src/controllers/messages/rooms.js

```diff
@@ -40,6 +42,49 @@ async function deactivateRoom(roomId) {
   return db.ref(`rooms/${roomId}/is_active`).set(false);
 }
 
+function handleTimers(listingId, roomId, from, to) {
+  // delay should actually be 48 hours in production, but is 30s for testing
+  const DELAY = 1000 * 30;
+
+  // we need to figure out if it's a buyer or if it's a seller
+  db.ref(`listings/${listingId}/seller`, async snapshot => {
+    const sellerId = snapshot.val();
+    let warningMessage, dangerMessage;
+    if (from == sellerId) {
+      // if the person sending the message is the seller
+      warningMessage = t.chat.buyer_warning;
+      dangerMessage = t.chat.buyer_danger;
+    } else {
+      // if the person sending the message is the buyer
+      warningMessage = t.chat.seller_warning;
+      dangerMessage = t.chat.seller_danger;
+    }
+
+    // we need to clear the stored timeouts
+    const lastMessageSnapshot = await db
+      .ref(`rooms/${roomId}/last_message`)
+      .once("value");
+    let { warning_timeout, timeout } = lastMessageSnapshot.val();
```

Comment:
`last_message` might be null here for the first message, make sure to add a check for that.

Reply:
Oh yeah I always forget about that 😩